src/main/java/dev/tourmi/svmm/server/VeinMiner.java

```diff
@@ -68,7 +70,8 @@ private boolean canTunnel(Player player, BlockState blockState) {
 
         if (!canMine(blockState, SVMMConfig.TUNNELING_WHITELIST.get(), SVMMConfig.TUNNELING_BLACKLIST.get())) {
             Tunneler.cancelTunnel(player.getUUID());
-            player.sendSystemMessage(Component.literal("Cancelled tunneling since the block that was mined is not part of the allowed list of blocks"));
+			ServerPlayer serverPlayer = (ServerPlayer) player; 
+            serverPlayer.sendSystemMessage(Component.literal("Cancelled tunneling since the block that was mined is not part of the allowed list of blocks"));
```

Comment:
You can revert this, updating the function signature will be enough to fix the compilation errors